src/Storage/Storage.Management/StorageAccount/SetAzureStorageAccount.cs

```diff
@@ -584,7 +583,33 @@ public override void ExecuteCmdlet()
 
             if (ShouldProcess(this.Name, "Set Storage Account"))
             {
-                if (this.force || this.AccessTier == null || ShouldContinue("Changing the access tier may result in additional charges. See (http://go.microsoft.com/fwlink/?LinkId=786482) to learn more.", ""))
+                string shouldContinueMessage = null;
+                if (this.UpgradeToStorageV2.IsPresent)
+                {
+                    if(this.OriginStorageAccountProperties.Kind == Kind.Storage)
+                    {
+                        shouldContinueMessage = "Upgrading a General Purpose v1 storage account to a general-purpose v2 account is free. You may specify the desired account tier during the upgrade process. If an account tier is not specified on upgrade, the default account tier of the upgraded account will be Hot. " + 
+                            "However, changing the storage access tier after the upgrade may result in changes to your bill so it is recommended to specify the new account tier during upgrade. See (http://go.microsoft.com/fwlink/?LinkId=786482) to learn more.";
+                    }
+                    else if (this.OriginStorageAccountProperties.Kind == Kind.BlobStorage)
+                    {
+                        shouldContinueMessage = "Upgrading a BlobStorage account to a general-purpose v2 account is free as long as the upgraded account�s tier remains unchanged. If an account tier is not specified on upgrade, the default account tier of the upgraded account will be Hot. " +
```

Comment:
Seems like something went wrong with the character encoding/decoding of the character here. It's showing accounts

Reply:
Thanks for the catch! 
Fixed.